src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientCleanup.kt

```diff
@@ -15,18 +17,20 @@ class RestrictedPatientCleanup(
 
   @Transactional
   fun deleteRestrictedPatientOnExternalMovementIntoPrison(prisonerNumber: String) {
-    val restrictedPatient = restrictedPatientsRepository.findByIdOrNull(prisonerNumber) ?: return
+    restrictedPatientsRepository.findByIdOrNull(prisonerNumber)?.let {
+      restrictedPatientsRepository.delete(it)
 
-    restrictedPatientsRepository.delete(restrictedPatient)
+      domainEventPublisher.publishRestrictedPatientRemoved(prisonerNumber)
 
-    domainEventPublisher.publishRestrictedPatientRemoved(prisonerNumber)
+      telemetryClient.trackEvent(
+        "restricted-patient-removed-cleanup",
+        mapOf("prisonerNumber" to prisonerNumber),
+        null
+      )
+    } ?: log.debug("Movement into prison for prisoner {} ignored as not a restricted patient", prisonerNumber)
```

Comment:
code improvement after last PR. I've removed the debug message to say that we're processing one of these events from the subscriber so can add message here